### PR TITLE
Fix patch validation agreements part 2

### PR DESCRIPTION
**Pull request by xlorepdarkhelm**
## What changed

There was further work needed to make PATCH functional.

## Issue

#1083

## How to test

`pytest`

There is a new test to validate the PATCH works.

## Screenshots

N/A

## Links

N/A

### CODE REVIEW

**Review by stevtek — APPROVED** [review]

File: backend/ops_api/tests/ops/agreement/test_agreement.py [hunk 1]
```diff
@@ -422,6 +422,22 @@ def test_agreements_patch_by_id_grant(auth_client, loaded_db):
...
+        "/api/v1/agreements/1",
+        json={
+            "notes": "Test PATCH",
+        },
+    )
+    assert response.status_code == 200
+
+    stmt = select(Agreement).where(Agreement.id == 1)
+    agreement = loaded_db.scalar(stmt)
+    assert agreement.notes == "Test PATCH"
```
Comment (tdonaworth) [inline backend/ops_api/tests/ops/agreement/test_agreement.py:438]: As indicated yesterday, do we want to clean up this record post PATCH?

Comment (xlorepdarkhelm) [reply backend/ops_api/tests/ops/agreement/test_agreement.py:438]: I can, I just copied what Steve had done, and didn't do anything else with it.

Comment (stevtek) [reply backend/ops_api/tests/ops/agreement/test_agreement.py:438]: I can work on this (after this PR) in my branch.
It's not just this one test but all the PUT, PATCH tests that I was thinking of creating a new record and deleting when done.

Comment (xlorepdarkhelm) [reply backend/ops_api/tests/ops/agreement/test_agreement.py]: @stevtek You can use what I did in that test as an example to get similar fixes in others if you want ;)

**Review by johndeange — APPROVED** [review]

**Review by tdonaworth — APPROVED** [review]

**Pull request merged**